SlackBotPrototype/API/MessageParser.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Console = System.Console;
+using edu.stanford.nlp.ling;
+using edu.stanford.nlp.tagger.maxent;
+
+namespace API
+{
+	public class MessageParser : IMessageParser
+	{
+		private readonly string _modelsDirectory;
+		
+		public MessageParser(string nlpFolder)
+		{
+			_modelsDirectory = nlpFolder + @"\models";
+		}
+
+		public string HandleRequest(string message)
+		{
+			var wordSet = BuildWordSet(message);
+			var weatherProvider = new DarkSkyWeatherProvider(ConfigConstants.DarkSkyApiSecret);
+			
+			if (wordSet.Contains("wear", StringComparer.InvariantCultureIgnoreCase) &&
+			    wordSet.Contains("I", StringComparer.InvariantCultureIgnoreCase))
+			{
+				var rawForecast = weatherProvider.GetForecastRaw(DateTime.Now);
+				var icon = rawForecast.Daily.Data.FirstOrDefault()?.Icon;
+				
+				// Skip accounting for temperature for now, could have thresholds for certain apparel
+				switch (icon)
+				{
+					case "clear-day":
+					case "clear-night":
+						return "Wear your comfortable clothing, no jacket required";
+					case "snow":
+						return "Wear winter jackets, bring boots!";
+					case "wind":
+						return "Bring a wind jacket, it\'s going to breezy";
+					case "partly-cloudy-day":
+					case "partly-cloudy-night":
+					case "cloudy":
+					case "fog":
+						return "Bring a jacket";
+					case "rain":
+						return "Dress comfortably, bring an umbrella";
+
+					default:
+						return "I don't know, anything you want?";
+				}
+			}
+
+			if (wordSet.Contains("weather", StringComparer.InvariantCultureIgnoreCase))
+			{
+				if (wordSet.Contains("now", StringComparer.InvariantCultureIgnoreCase)
+					|| wordSet.Contains("today", StringComparer.InvariantCultureIgnoreCase))
+				{
+					return weatherProvider.GetForecast(DateTime.Now);
+				}
+
+				if (wordSet.Contains("tomorrow"))
+				{
+					return weatherProvider.GetForecast(DateTime.Now.AddDays(1));
+				}
+
+				return "Weather now or tomorrow?";
+			}
+
+			return string.Empty;
+		}
+
+		public HashSet<string> BuildWordSet(string message)
+		{
+			var tagger = new MaxentTagger(_modelsDirectory + @"\wsj-0-18-bidirectional-distsim.tagger");

```

Comment:
Good catch, definitely an optimization that should be made. Should initialize the tagger in the constructor.